### ISSUE TITLE
Compile error with GHC 7.4.2

### ISSUE THREAD

**Issue by hvr** [nh2/haskell-netpbm#3]
I noticed the following TH/compile error with GHC 7.4:

```
Configuring component lib from netpbm-1.0.2
Preprocessing library netpbm-1.0.2...
[1 of 1] Compiling Graphics.Netpbm  ( src/Graphics/Netpbm.hs, /tmp/matrix-worker/1480326130/dist-newstyle/build/x86_64-linux/ghc-7.4.2/netpbm-1.0.2/build/Graphics/Netpbm.o )
Loading package ghc-prim ... linking ... done.
Loading package integer-gmp ... linking ... done.
Loading package base ... linking ... done.
Loading package array-0.4.0.0 ... linking ... done.
Loading package deepseq-1.3.0.0 ... linking ... done.
Loading package containers-0.4.2.1 ... linking ... done.
Loading package pretty-1.1.1.0 ... linking ... done.
Loading package template-haskell ... linking ... done.
Loading package transformers-0.5.2.0 ... linking ... done.
Loading package primitive-0.6.1.0 ... linking ... done.
Loading package vector-0.11.0.0 ... linking ... done.
Loading package vector-th-unbox-0.2.1.6 ... linking ... done.
Loading package utility-ht-0.0.12 ... linking ... done.
Loading package storable-record-0.0.3.1 ... linking ... done.
Loading package bytestring-0.9.2.1 ... linking ... done.
Loading package fail-4.9.0.0 ... linking ... done.
Loading package binary-0.5.1.0 ... linking ... done.
Loading package bytestring-builder-0.10.8.1.0 ... linking ... done.
Loading package text-1.2.2.1 ... linking ... done.
Loading package hashable-1.2.4.0 ... linking ... done.
Loading package scientific-0.3.4.9 ... linking ... done.
Loading package nats-1.1.1 ... linking ... done.
Loading package transformers-compat-0.5.1.4 ... linking ... done.
Loading package tagged-0.8.5 ... linking ... done.
Loading package unordered-containers-0.2.7.1 ... linking ... done.
Loading package semigroups-0.18.2 ... linking ... done.
Loading package attoparsec-0.13.1.0 ... linking ... done.
Loading package attoparsec-binary-0.2 ... linking ... done.

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeFreeze' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeThaw' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `basicLength' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeSlice' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeIndexM' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeCopy' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `elemseq' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeFreeze' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeThaw' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `basicLength' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeSlice' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeIndexM' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeCopy' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `elemseq' is not a (visible) method of class `Data.Vector.Generic.Base.Vector'

src/Graphics/Netpbm.hs:132:1:
    `basicLength' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeSlice' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicOverlaps' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeNew' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicInitialize' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeReplicate' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeRead' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeWrite' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicClear' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicSet' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeCopy' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeMove' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeGrow' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicLength' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeSlice' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicOverlaps' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeNew' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicInitialize' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeReplicate' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeRead' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeWrite' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicClear' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicSet' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeCopy' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeMove' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

src/Graphics/Netpbm.hs:132:1:
    `basicUnsafeGrow' is not a (visible) method of class `Data.Vector.Generic.Mutable.Base.MVector'

```

**Comment by hvr** [comment]
The problem appears to be that GHC 7.4.2 needs special treatment as described in https://hackage.haskell.org/package/vector-th-unbox-0.2.0.2/docs/Data-Vector-Unboxed-Deriving.html

Adding the imports

```hs
import qualified Data.Vector.Generic
import qualified Data.Vector.Generic.Mutable
```

fixes compilation w/ GHC 7.4.2;

I went ahead and fixed up the meta-data on Hackage to exclude GHC 7.4.2 for netpbm-1.0.2 (netpbm-1.0.1 seems to work fine w/ GHC 7.4):

https://hackage.haskell.org/package/netpbm-1.0.2/revisions/

**Comment by hvr** [comment]
I just noticed, that netpbm-1.0.1 fixed this very issue (as it existed already in 1.0.0), but for some reason those imports went missing again for the 1.0.2 release via 992698f314e80d1158ddcc473e77c2e92b8dd688

**Comment by lehins** [comment]
@hvr, I believe, because there was no ghc version check `#if __GLASGOW_HASKELL == 704`, it was causing generation of some warnings on newer ghc versions, which is probably why it resulted in removal of those imports in 1.0.2

**Comment by hvr** [comment]
@lehins sounds plausible; btw I noticed this problem because it was causing indirect failures for `hip`...

**Comment by lehins** [comment]
@hvr I figured that :) Thank you for going the distance. Btw, I was the one who added those imports couple years ago.

**Comment by nh2** [comment]
I've included the base-4.5 exclude in https://hackage.haskell.org/package/netpbm-1.0.3
